### ISSUE TITLE
Accessibility: screen reader isn't reading modal correctly

### ISSUE THREAD

**Issue by halimahLDN** [Greater-London-Authority/ldn-viz-tools#831]
**Describe the bug**
Currently the screen reader can't read the Modal contents when it is first opened. Whatever receives focus is read out but not the whole content.

The default focus is the 'Close' button which, if focused, will be dictated but nothing else will be read out. If the title is focused, only the title and close button will be read out.

**To Reproduce**
Using VoiceOver, navigate to the Modal button in Storybook, open it and witness it!

You can also test on this simplified version [I made on StackBlitz](https://stackblitz.com/edit/github-t7odbkuc?file=src%2Flib%2Fcomponents%2FDialog.svelte). Note: the code is taken from the Melt docs.

**Expected behavior**
- Upon opening the Modal should either focus on the first interactive element or the first element (e.g. title).
- Using VoiceOver controls (`Control + Option + A`), you should be able to get the screen reader to read out the whole modal.

**Screenshots**
In Storybook, when the focus is set to the title, it will read the title, close button and then contents outside of the frame.

In a real-life application (apps/notebooks), it won't read outside of the modal but gets stuck on the close button:

https://github.com/user-attachments/assets/79b5c66f-d15d-41b0-8ff7-5cd8572c244c

**Desktop (please complete the following information):**

- OS: iOS
- Browser Chrome
- Version 132

**Additional context**
I did try to change the focus to `$content.id` but this is not accessible when modals have long content as the whole content of the modal is read out in one string, so WAI ARIA recommends against this in the [modal design pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/).
